src/Libraries/PythonNodeModels/PythonNode.cs

```diff
@@ -47,7 +48,7 @@ public PythonEngineVersion Engine
                 // This is a first-time case for newly created nodes only
                 if (engine == PythonEngineVersion.Unspecified)
                 {
-                    SetEngineByDefault();
+                    SetEngineByDefault(ref engine);
```

Comment:
I'm very confused why you would use a `ref` parameter here.

Reply:
I am using the same function to update different instances of PythonEngineVersion, therefore passing by reference.

Reply:
I am sorry, I just saw that I am no longer doing that, will update this